impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageFromWarTest.java

```diff
@@ -0,0 +1,102 @@
+package org.jboss.shrinkwrap.impl.base.spec;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+
+import junit.framework.Assert;
+import org.jboss.shrinkwrap.api.ConfigurationBuilder;
+import org.jboss.shrinkwrap.api.Domain;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.impl.base.spec.donotchange.DummyClassA;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Bob McWhirter
+ */
+public class AddPackageFromWarTest {
+
+    private Domain domain;
+
+    private Path temp;
+
+    @Before
+    public void setUp() throws IOException {
+
+        ConfigurationBuilder builder = new ConfigurationBuilder();
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "my.war");
+        war.addClass(DummyClassA.class);
+
+        this.temp = Files.createTempFile("test", ".war");
+        war.as(ZipExporter.class).exportTo(temp.toFile(), true);
+        URL warUrl = temp.toUri().toURL();
+
+        List<ClassLoader> classLoaders = new ArrayList<ClassLoader>();
+
+        URLClassLoader warCl = new URLClassLoader(new URL[]{warUrl}, null);
```

Comment:
We have strange classloaders in Swarm land. 

--
Sent from my iPhone


> On May 31, 2016, at 5:36 PM, Andrew Lee Rubinger <notifications@github.com> wrote:
> 
> In impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageFromWarTest.java:
> 
> > +    private Path temp;
> > +
> > +    @Before
> > +    public void setUp() throws IOException {
> > +
> > +        ConfigurationBuilder builder = new ConfigurationBuilder();
> > +        WebArchive war = ShrinkWrap.create(WebArchive.class, "my.war");
> > +        war.addClass(DummyClassA.class);
> > +
> > +        this.temp = Files.createTempFile("test", ".war");
> > +        war.as(ZipExporter.class).exportTo(temp.toFile(), true);
> > +        URL warUrl = temp.toUri().toURL();
> > +
> > +        List<ClassLoader> classLoaders = new ArrayList<ClassLoader>();
> > +
> > +        URLClassLoader warCl = new URLClassLoader(new URL[]{warUrl}, null);
> Ah the no-parent CL, cute. Nice stuff.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>